apps/backoffice-v2/src/domains/business-reports/hooks/mutations/useCreateBusinessReportBatchMutation/useCreateBusinessReportBatchMutation.tsx

```diff
@@ -0,0 +1,45 @@
+import { t } from 'i18next';
+import { toast } from 'sonner';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+
+import { HttpError } from '@/common/errors/http-error';
+import { TBusinessReportType } from '@/domains/business-reports/types';
+import { createBatchBusinessReport } from '@/domains/business-reports/fetchers';
+import { useCustomerQuery } from '@/domains/customer/hook/queries/useCustomerQuery/useCustomerQuery';
+
+export const useCreateBusinessReportBatchMutation = ({
+  reportType,
+  onSuccess,
+}: {
+  reportType: TBusinessReportType;
+  onSuccess?: <TData>(data: TData) => void;
+}) => {
+  const queryClient = useQueryClient();
+
+  const { data: customer } = useCustomerQuery();
+
+  return useMutation({
+    mutationFn: (merchantSheet: string) =>
+      createBatchBusinessReport({
+        reportType,
+        merchantSheet,
+        isExample: customer?.config?.isExample ?? false,
+      }),
+    onSuccess: data => {
+      void queryClient.invalidateQueries();
```

Comment:
general question: what this line is actually does ? why do you call it on success ?

Reply:
It causes invalidation of the `react-query`'s cache so when we go to the Reports page it would trigger a fetch request for all reports and we could see the new ones. Without this line, we will not see the new reports before refreshing or cache TTL expiry

Reply:
It refetches every query we have in the backoffice to ensure data that depends on other queries is up-to-date and reflects what this mutation changed. Cherry picking what query changed is not worth it today in our context.